src/platform/backends/shared/base_availability_zone.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "multipass/base_availability_zone.h"
+#include "multipass/exceptions/availability_zone_exceptions.h"
+#include "multipass/file_ops.h"
+#include "multipass/logging/log.h"
+
+#include <fmt/format.h>
+
+#include <QJsonDocument>
+
+namespace multipass
+{
+
+namespace mpl = logging;
+
+constexpr auto subnet_key = "subnet";
+constexpr auto available_key = "available";
+
+BaseAvailabilityZone::data BaseAvailabilityZone::make(const std::string& name, const fs::path& az_directory)
+{
+    const auto file_path{az_directory / (name + ".json")};
+
+    mpl::log(mpl::Level::info, name, "creating zone");
+
+    std::error_code err;
+    const auto file_type = MP_FILEOPS.status(file_path, err).type();
+    if (err && file_type != fs::file_type::not_found)
+    {
+        throw AvailabilityZoneDeserializationError{
+            "AZ file {:?} is not accessible: {}.",
+            file_path.string(),
+            err.message(),
+        };
+    }
+
+    if (file_type == fs::file_type::not_found)
+    {
+        mpl::log(mpl::Level::info, name, fmt::format("AZ file {:?} not found, using defaults", file_path.string()));
+        return {
+            .name = name,
+            .file_path = file_path,
+            // TODO GET ACTUAL SUBNET
+            .subnet = "",
+            .available = true,
```

Comment:
Wait, how come we have designated initializers... 🤔

Reply:
Good catch. GCC and clang seem to be lax about this unless compiled with `-pedantic`: https://gcc.godbolt.org/z/j6xMYscnj. Even with `-pedantic', it's a warning. MSVC throws a hard error without any need for flags, though.

Reply:
:sob:
I could do it like this instead
```cpp
return {
    /*.name = */ name,
    /*.file_path = */ file_path,
    // TODO GET ACTUAL SUBNET
    /*.subnet = */ "",
    /*.available = */ true,
};
```
to at least clarify where the values go, although that could give a false sense of security...

Reply:
I am fine with that. You could also say "uncomment in C++20".